refactor/tables to complete associations, feature/package routes

alantly opened the pull request.

Commit:
refactored routes to be more clear per route file

alantly commented:
Do not merge yet. We need to merge the package migration and model first.

Commit:
routes for packages

supaflyqtpie commented on app.js:
```diff
@@ -56,13 +56,15 @@ initPassport(passport);
 // Setup server API routes
-app.use('/api/user', user);
-app.use('/api/session', userSession);
+app.use('/api', user);
```
Is there any specific motivation for this refactoring?

Commit:
Merge branch 'master' into feature/package-routes

Commit:
mocha working on failing test

Commit:
updated package and readme to shortcut db commands

added test env variable to use test db

setup a more complete test and moved sequelize sync into utils

update readme

association and test for package and user created

cleanup test name

Commit:
clean up second before each

Commit:
Merge branch 'bug/mocha-with-models' into feature/package-routes

Commit:
add user spec

Commit:
set up association test and model tests for user functions, renamed package specs, turned off sequelize logging for test

Commit:
Merge pull request #38 from supaflyqtpie/feature/user-unit-tests

Feature/user unit tests

Commit:
recreate user model using sequelize generator

Commit:
update package and package histories

Commit:
update seed files

Commit:
updated packages to use associations

Commit:
Merge branch 'bug/mocha-with-models' into feature/package-routes
